packages/neon_framework/lib/src/storage/persistence.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart' show protected;
+
+/// A persistent key value storage.
+abstract interface class Persistence<T extends Object> {
+  /// Whether a value exists at the given [key].
+  FutureOr<bool> containsKey(String key);
+
+  /// Completes with true once all values have been cleared.
+  FutureOr<bool> clear();
+
+  /// Removes an entry from persistent storage.
+  FutureOr<bool> remove(String key);
+
+  /// Saves a [value] to persistent storage.
+  FutureOr<bool> setValue(String key, T value);
+
+  /// Fetches a the value persisted at the given [key] from the persistent
```

Comment:
```suggestion
  /// Fetches a the value persisted with the given [key] from the persistent
```

Reply:
I think the at is better because a value is stored at a key (location) and not with a key but I don't really care.

Reading this doc comment I find the start more interesting ` Fetches a the value` :upside_down_face: